app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Validation\ValidationException;
+
+class AuthController extends Controller
+{
+    public function login(Request $request) {
+        $request->validate([
+            'email' => 'required|email',
+            'password' => 'required',
+            'device_name' => 'required',
+        ]);
```

Comment:
I think we need to create a new file for the `request` Validator.

Reply:
why?

Reply:
To validate whether this field is filled or not. And a little code to keep its cleanup. I think we need to use the `Single Responsibility Principle` here.